File: js/src/maplestory/api/kms/dto/union/union.ts
import * as base from '../../../common/dto/union/union';
import { UnionBody } from '../../response/union/unionBody';

/**
 * 유니온 정보
 */
export class UnionDto extends base.UnionDto {
  /**
   * 조회 기준일
   */
  public override date: Date | null;

  /**
   * 유니온 레벨
   */
  public override unionLevel: number | null;

  /**
   * 유니온 등급
   */
  public override unionGrade: string | null;

  /**
   * 아티팩트 레벨
   */
  public override unionArtifactLevel: number | null;

  /**
   * 보유 아티팩트 경험치
   */
  public override unionArtifactExp: number | null;

  /**
   * 보유 아티팩트 포인트
   */
  public override unionArtifactPoint: number | null;

  constructor(obj: UnionBody) {
    super();

    const {
      date,
      union_level,
      union_grade,
      union_artifact_level,
      union_artifact_exp,
      union_artifact_point,
    } = obj;

    this.date = date ? new Date(date) : null;
    this.unionLevel = union_level;
    this.unionGrade = union_grade;
    this.unionArtifactLevel = union_artifact_level;
    this.unionArtifactExp = union_artifact_exp;
    this.unionArtifactPoint = union_artifact_point;
  }
}

File: csharp/MapleStory.OpenAPI/Src/KMS/Dto/Union/Union.cs
using Newtonsoft.Json;
using Base = MapleStory.OpenAPI.Common.DTO;

namespace MapleStory.OpenAPI.KMS.DTO
{
    /// <summary>
    /// 유니온 정보
    /// </summary>
    public class UnionDTO : Base.UnionDTO
    {
        /// <summary>
        /// 조회 기준일 (KST, 일 단위 데이터로 시, 분은 일괄 0으로 표기)
        /// </summary>
        [JsonProperty("date")]
        public override DateTimeOffset? Date
        {
            get
            {
                return _date?.ToOffset(TimeSpan.FromHours(9));
            }
            set
            {
                _date = value;
            }
        }

        private DateTimeOffset? _date;

        /// <summary>
        /// 유니온 레벨
        /// </summary>
        [JsonProperty("union_level")]
        public override int? UnionLevel { get; set; }

        /// <summary>
        /// 유니온 등급
        /// </summary>
        [JsonProperty("union_grade")]
        public override string? UnionGrade { get; set; }

        /// <summary>
        /// 아티팩트 레벨
        /// </summary>
        [JsonProperty("union_artifact_level")]
        public override int? UnionArtifactLevel { get; set; }

        /// <summary>
        /// 보유 아티팩트 경험치
        /// </summary>
        [JsonProperty("union_artifact_exp")]
        public override int? UnionArtifactExp { get; set; }

        /// <summary>
        /// 보유 아티팩트 포인트
        /// </summary>
        [JsonProperty("union_artifact_point")]
        public override int? UnionArtifactPoint { get; set; }
    }
}
